### ISSUE TITLE
support unsigned numeric tags?

### ISSUE THREAD

**Issue by ch-yx** [vberlier/nbtlib#29]
Could you please support unsigned numeric tags?

- [x]  `lambda self:self&(self.range.stop-self.range.start-1)`

**Comment by vberlier** [comment]
I'm not sure what you mean by unsigned numeric tags, as far as I know all integer tags are signed. Do you have a specific example in mind?

**Comment by ch-yx** [comment]
Add something like that to numberic tag:
```
@property 
def unsign (self):
 return self&(len(self.range)-1)

```
and 
```
@classmethod
def from_unsign(c,x):
 range =len(c.range)
 x=x%range
 return c(x-(x not in c.range)*range)
```

**Comment by vberlier** [comment]
Hmm okay so it would allow you to interpret the tag as an unsigned integer. I don't know if there's a real use-case but I think I would just prefer to create my own userland function for this.

**Comment by ch-yx** [comment]
There is.
For example,If you want to use UUIDLeast/Most and string UUID 😏

**Comment by vberlier** [comment]
Wait UUIDLeast/Most is not just a long tag?

**Comment by ch-yx** [comment]
I'm not saying it is just a long tag.
But meaning of these two tag is unsigned.
Sometimes you have to connect them together or convert them to or from String form(xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx).
It will be neat if we have these two helper method.

-----------------



By the way,I will try to not use any "magic" thing to solve that without slowness.

**Comment by vberlier** [comment]
Hmm okay I think I understand. I guess adding these two helpers won't cost much anyway. It would be cool to also add a working example showing how to use them to combine and split a UUID string.

```py
# examples/uuid.py

def combine_uuid(uuid_most_tag, uuid_least_tag):
    uuid_most = uuid_most_tag.as_unsigned
    uuid_least = uuid_least_tag.as_unsigned
    return ...

def split_uuid(uuid_string_tag):
    uuid_most = ...
    uuid_least = ...
    return Long.from_unsigned(uuid_most), Long.from_unsigned(uuid_least)
```

**Comment by ch-yx** [comment]
by the way,I'm now starting to try to implement NBT Path.
https://minecraft.gamepedia.com/Commands/data#NBT_path
I'm planning to make it like Pathlib's Path

**Comment by vberlier** [comment]
Just added the unsigned helpers with the uuid example I talked about 74a802829b786070a5419deaab759ac63bc3285d.

Nbt paths could be a pretty useful feature, but we should probably discuss that in a separate issue.

**Comment by vberlier** [comment]
Okay just released v1.5.0, closing the issue.
